pkg/webhook/cvc.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2020 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"encoding/json"
+	"net/http"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	clientset "github.com/openebs/maya/pkg/client/generated/clientset/versioned"
+	cvc "github.com/openebs/maya/pkg/cstorvolumeclaim/v1alpha1"
+	util "github.com/openebs/maya/pkg/util"
+	"github.com/pkg/errors"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+type validateFunc func(cvcOldObj, cvcNewObj *apis.CStorVolumeClaim) error
+
+type getCVC func(name, namespace string, clientset clientset.Interface) (*apis.CStorVolumeClaim, error)
+
+func (wh *webhook) validateCVCUpdateRequest(req *v1beta1.AdmissionRequest, getCVC getCVC) *v1beta1.AdmissionResponse {
+	response := NewAdmissionResponse().
+		SetAllowed().
+		WithResultAsSuccess(http.StatusAccepted).AR
+	var cvcNewObj apis.CStorVolumeClaim
+	err := json.Unmarshal(req.Object.Raw, &cvcNewObj)
```

Comment:
for this to work, upgrade of CVC should have been done?

Reply:
Yes!! The user/Controller made an update call on CVC.